src/executor/executor.py

```diff
@@ -0,0 +1,429 @@
+                                   'action'  : rp.LINK}]
+            cud.cores          = self._DEMcores
+            cud.mpi            = True
+
+            # Submit the first unit and wait until it staged its input files. Waiting is
+            # needed so that we can get the path of the unit and pass it to the DEM monitor.
+            dem_unit = self._umgr.submits_units(cud)
+            # This line blocks the execution until the DEM unit has a path.
+            self._umgr.wait_units(uids=dem_unit.uid,state=rp.states.AGENT_SCHEDULING_PENDING)
+
+            #Now that we have a path we can continue
+            cud2 = rp.ComputeUnitDescription()
+            cud2.executable = 'python'
+            cud2.arguments = ['controller_DEMresource_main.py',timestep,types,ru.Url(dem_unit.sandbox).path]
+            cud2.input_staging = [{'source':'file:///controller_DEMresource_main.py'
+                                  'target':'unit:///controller_DEMresource_main.py'
+                                  'action': rp.TRANSFER},
+                                  {'source':'file:///controller_DEMresource_data_reader.py'
+                                  'target':'unit:///controller_DEMresource_data_reader.py'
+                                  'action': rp.TRANSFER},
+                                  {'source':'file:///controller_DEMresource_data_interpretor.py'
+                                  'target':'unit:///controller_DEMresource_data_interpretor.py'
+                                  'action': rp.TRANSFER}]
+            cud2.output_staging = [{'source': 'unit:///PBM_input.json',
+                                   'target': 'file:///PBM_input.json',
+                                   'action'  : rp.TRANSFER},
+                                   {'source': 'unit:///DEM_status.json',
+                                   'target': 'file:///DEM_status.json',
+                                   'action'  : rp.TRANSFER}]
+
+            # Submit the monitor unit and return
+            dem_monitor_unit = self._umgr.submits_units(cud2)
+
+
+            self._dem_unit = dem_unit
+            self._dem_monitor_unit = dem_monitor_unit
+
+    def _check_DEM_status(self,status_file):
+        """
+        This method checks the state of the DEM execution and based on the status code
+        reports whether or not the simulation should continue with  starting PBM or 
+        stop everything.
+
+        Inputs:
+        status_file: The name of a json file that contains the status and other entries
+                     necessary for the PBM execution
+
+        Returns:
+        cont: A boolean value that shows if the simulation should continue or not
+        arguments: A list that contains the necessary arguments for the PBM execution.
+        """
+
+        status_fid = open(status_file)
+        status_dict = json.load(status_fid)
+        status_fid.close()
+
+        if status_dict['status'] == 1:
+            cont = True
+        else:
```

Comment:
we will to add an elif for status as 2 as well. This is where we kill all executions